helix-term/src/commands.rs

```diff
@@ -1875,6 +1876,27 @@ fn extend_line(cx: &mut Context) {
     doc.set_selection(view.id, selection);
 }
 
+fn extend_line_above(cx: &mut Context) {
+    let count = cx.count();
+    let (view, doc) = current!(cx.editor);
+
+    let text = doc.text();
+    let selection = doc.selection(view.id).clone().transform(|range| {
+        let (start_line, end_line) = range.line_range(text.slice(..));
+
+        let mut start = text.line_to_char(start_line);
+        let end = text.line_to_char((end_line + count).min(text.len_lines()));
+
+        // go to previous line if current line is selected
+        if range.from() == start && range.to() == end {
+            start = text.line_to_char(start_line.saturating_sub(1));
+        }
+        Range::new(end, start)
+    });
+
+    doc.set_selection(view.id, selection);
+}
```

Comment:
This is a lot of duplication for essentially changing two lines. Can you factor out the common implementation in a similar way we do for other commands like https://github.com/helix-editor/helix/blob/b6e452151196bf773874d266f9bf82e7386e56ef/helix-term/src/commands.rs#L4349-L4380=